src/FlatFeeCalculator.sol

```diff
@@ -0,0 +1,179 @@
+        for (uint256 i = 0; i < recipientsLength; i++) {
+            shares[i] = (totalFee * _shares[i]) / 100;
+            restFee -= shares[i];
+        }
+
+        // If any fee is left, it is distributed to the first recipient.
+        // This may happen if any of the shares of the fee to be distributed
+        // has leftover from the division by 100 above.
+        shares[0] += restFee;
+
+        feeDistribution.recipients = _recipients;
+        feeDistribution.shares = shares;
+    }
+
+    /// @notice Calculates the redemption fees for a given amount.
+    /// @param pool The address of the pool.
+    /// @param tco2s The addresses of the TCO2 token.
+    /// @param redemptionAmounts The amounts to be redeemed.
+    /// @return feeDistribution How the fee is meant to be
+    /// distributed among the fee recipients.
+    function calculateRedemptionFees(address pool, address[] calldata tco2s, uint256[] calldata redemptionAmounts)
+        external
+        view
+        override
+        returns (FeeDistribution memory feeDistribution)
+    {
+        require(tco2s.length == redemptionAmounts.length, "length mismatch");
+        require(tco2s.length == 1, "only one");
+
+        feeDistribution = _calculateFee(redemptionAmounts[0]);
+    }
+
+    /// @notice Calculates the deposit fee for a given amount of an ERC1155 project.
+    /// @param pool The address of the pool.
+    /// @param erc1155 The address of the ERC1155 project
+    /// @param tokenId The tokenId of the vintage.
+    /// @param depositAmount The amount to be deposited.
+    /// @return feeDistribution How the fee is meant to be
+    /// distributed among the fee recipients.
+    function calculateDepositFees(address pool, address erc1155, uint256 tokenId, uint256 depositAmount)
+        external
+        view
+        override
+        returns (FeeDistribution memory feeDistribution)
+    {
+        require(depositAmount > 0, "depositAmount must be > 0");
+
+        feeDistribution = _calculateFee(depositAmount);
+    }
+
+    /// @notice Calculates the redemption fees for a given amount on ERC1155 projects.
+    /// @param pool The address of the pool.
+    /// @param erc1155s The addresses of the ERC1155 projects.
+    /// @param tokenIds The tokenIds of the project vintages.
+    /// @param redemptionAmounts The amounts to be redeemed.
+    /// @return feeDistribution How the fee is meant to be
+    /// distributed among the fee recipients.
+    function calculateRedemptionFees(
+        address pool,
+        address[] calldata erc1155s,
+        uint256[] calldata tokenIds,
+        uint256[] calldata redemptionAmounts
+    ) external view override returns (FeeDistribution memory feeDistribution) {
+        require(erc1155s.length == tokenIds.length, "erc1155s/tokenIds length mismatch");
+        require(erc1155s.length == redemptionAmounts.length, "erc1155s/redemptionAmounts length mismatch");
+        require(erc1155s.length == 1, "only one");
```

Comment:
Technically here we don't need to restrict to a single vintage since the fee is flat so it's easy to calculate for multiple.